Tests/tests/test_Warehouse.cpp

```diff
@@ -0,0 +1,803 @@
+                return WDMP_SUCCESS;
+            }));
+
+    //Invoke executeHardwareTest
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("executeHardwareTest"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+}
+
+TEST_F(WarehouseInitializedTest, getHardwareTestResults)
+{
+    EXPECT_CALL(rfcApiImplMock, getRFCParameter(::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char* pcCallerID, const char* pcParameterName, RFC_ParamData_t* pstParamData) {
+                EXPECT_EQ(string(pcCallerID), string("Warehouse"));
+                EXPECT_EQ(string(pcParameterName), string("Device.DeviceInfo.X_RDKCENTRAL-COM_xOpsDeviceMgmt.hwHealthTest.Results"));
+                strncpy(pstParamData->value, "test", sizeof(pstParamData->value));
+                return WDMP_SUCCESS;
+            }));
+    EXPECT_CALL(rfcApiImplMock, setRFCParameter(::testing::_, ::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [](char* pcCallerID, const char* pcParameterName, const char* pcParameterValue, DATA_TYPE eDataType) {
+                EXPECT_EQ(string(pcCallerID), _T("Warehouse"));
+                EXPECT_EQ(string(pcParameterName), _T("Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.hwHealthTest.Enable"));
+                EXPECT_EQ(string(pcParameterValue), _T("false"));
+                EXPECT_EQ(eDataType, WDMP_BOOLEAN);
+                return WDMP_SUCCESS;
+            }));
+
+    //Invoke getHardwareTestResults
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getHardwareTestResults"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"testResults\":\"test\",\"success\":true}"));
+}
+
+TEST_F(WarehouseInitializedTest, getDeviceInfo)
+{
+    const string deviceInfoFile = _T("/tmp/deviceInfoTemp.txt");
+    const uint8_t deviceInfoContent[] = "bluetooth_mac=12:34:56:78:90:AB\n"
+                                    "boxIP=192.168.1.0\n"
+                                    "build_type=VBN\n"
+                                    "estb_mac=12:34:56:78:90:AB\n"
+                                    "eth_mac=12:34:56:78:90:AB\n"
+                                    "friendly_id=Abc XYZ\n"
+                                    "imageVersion=ABCADS_VBN_2022010101sdy__MH_SPLASH_TEST_2\n"
+                                    "model_number=ABC123ADS\n"
+                                    "wifi_mac=12:34:56:78:90:AB\n";
+
+    ON_CALL(wrapsImplMock, popen(::testing::_, ::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [&](const char* command, const char* type) -> FILE* {
+                return fopen(deviceInfoFile.c_str(), type);
+            }));
+
+    //Invoke getDeviceInfo
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getDeviceInfo"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
```

Comment:
since "getDeviceInfo" is expected to call "sh /lib/rdk/getDeviceDetails.sh read" please write a fake /lib/rdk/getDeviceDetails.sh that will return the response

Reply:
Done